python/tests/test_highlevel.py

```diff
@@ -1591,7 +1619,12 @@ def test_tree_node_edges(self):
             for mapping, tree in zip(ts._tree_node_edges(), ts.trees()):
                 node_mapped = mapping >= 0
                 edge_visited[mapping[node_mapped]] = True
-                assert np.sum(node_mapped) == len(list(tree.nodes())) - tree.num_roots
+                # Paradoxically, tree.nodes() does not necessarily list all the nodes
```

Comment:
I don't think it's a paradox, it naturally follows from the fact that it lists nodes in traversal order. The documentation isn't clear about this all right though; I've opened #1285

Reply:
Yeah, I commented there too. Happy to rephrase this comment